src/kicadtoNgspice/DeviceModel.py

```diff
@@ -198,162 +233,187 @@ def __init__(self, schematicInfo, clarg1):
+                self.count = self.count + 1
+
+                self.lengthLabel[self.count] = QtGui.QLabel(
+                    "Enter length of MOSFET " + words[0] + "(default=100u):")
                 mosfetgrid.addWidget(self.lengthLabel[self.count], self.row, 0)
                 self.entry_var[self.count] = QtGui.QLineEdit()
                 self.entry_var[self.count].setText("")
                 self.entry_var[self.count].setMaximumWidth(150)
                 mosfetgrid.addWidget(self.entry_var[self.count], self.row, 1)
                 self.row = self.row + 1
-                self.count = self.count+1
-                
-                
-                self.multifactorLable[self.count] = QtGui.QLabel("Enter multiplicative factor of MOSFET "+words[0]+"(default=1):")
-                mosfetgrid.addWidget(self.multifactorLable[self.count], self.row, 0)
+                self.count = self.count + 1
+
+                self.multifactorLable[self.count] = QtGui.QLabel(
+                    "Enter multiplicative factor of MOSFET " +
+                    words[0] + "(default=1):")
+                mosfetgrid.addWidget(
+                    self.multifactorLable[self.count], self.row, 0)
                 self.entry_var[self.count] = QtGui.QLineEdit()
                 self.entry_var[self.count].setText("")
                 end = self.count
                 self.entry_var[self.count].setMaximumWidth(150)
                 mosfetgrid.addWidget(self.entry_var[self.count], self.row, 1)
                 self.row = self.row + 1
                 self.devicemodel_dict_end[words[0]] = self.count
-                self.count = self.count+1
+                self.count = self.count + 1
                 mosfetbox.setLayout(mosfetgrid)
-                #global path_name
+                # global path_name
                 try:
                     for key in json_data["deviceModel"]:
                         if key[0] == eachline[0] and key[1] == eachline[1]:
-                            #print "DEVICE MODEL MATCHING---",child.tag[0],child.tag[1],eachline[0],eachline[1]
+                            # print "DEVICE MODEL MATCHING---",child.tag[0],\
+                            # child.tag[1],eachline[0],eachline[1]
                             while i <= end:
-                                self.entry_var[i].setText(json_data["deviceModel"][key][i-beg])
-                                if (i-beg) == 0:
-                                    if os.path.exists(json_data["deviceModel"][key][0]):
-                                        path_name = json_data["deviceModel"][key][0]
+                                self.entry_var[i].setText(
+                                    json_data["deviceModel"][key][i - beg])
+                                if (i - beg) == 0:
+                                    if os.path.exists(
+                                            json_data["deviceModel"][key][0]):
+                                        path_name = json_data
```

Comment:
need backslash at the end for continuation

Reply:
Sir, it works properly without backslash as well, tested it in a mock file as -
```
import os
if os.path.exists(
    "."):
    print("works")
```

The above code executes successfully.
Is backslash to be added just as a syntactic sugar?

Reply:
Problem is not in the if block, but the line after that. It should be:

```
path_name = json_data\
      ["deviceModel"][key][0]
```

Reply:
different ways to do this (anything is okay for me):

```
path_name = json_data\
      ["deviceModel"][key][0]

```

```
path_name = (json_data
      ["deviceModel"][key][0])

```

```
path_name = json_data[
      "deviceModel"][key][0]

```

Reply:
Thanks,
Resolved with the latest commit on documentation branch.

Ref - #86